transport/http/outbound.go

```diff
@@ -399,6 +406,12 @@ func getYARPCErrorFromResponse(response *http.Response, bothResponseError bool)
 	)
 }
 
+// Only does verification if there is a response header
+func checkServiceMatch(reqSvcName string, resHeaders http.Header) (bool, string) {
+	serviceName := resHeaders.Get(ServiceHeader)
+	return serviceName == "" || serviceName == reqSvcName, serviceName
```

Comment:
This is permissive of empty service header - probably the right default choice now, but we might want to allow callers to choose an enforcement policy (as in don't process responses without them).

Reply:
Thanks for your suggestion, Zack. I'm not sure with logic here, and I need to confirm with other team members, @AlexAPB @abhinav . Personally, I think the logic here is more like a defensive programming instead of a configurable option for caller. We originally put this default check since we deploy our service in an iterative way, some clients' requests may hit the server that use the old version.

Reply:
This is essentially a safe guard for us to make sure that YARPC is doing the right thing. Instead of making users opt-in, I'd rather it be a transparent step to ensure we're communicating properly.

Reply:
> This is essentially a safe guard for us to make sure that YARPC is doing the right thing. Instead of making users opt-in, I'd rather it be a transparent step to ensure we're communicating properly.

Mostly agreed. See detailed response above.